### DISCUSSION TITLE
HTML parsing in  Templater

### DISCUSSION THREAD

**Discussion by Smitty010** [SilentVoid13/Templater#746] [Help]
Is there an html parser that is available when I'm running a templater script?  I want to take an html file in my vault and convert it to markdown based on custom rules.  I can probably use regex, but a real html parser seems cleaner.

**Comment by sohanglal** [comment]
- [htmlToMarkdown | Obsidian Plugin Developer Docs](https://marcus.se.net/obsidian-plugin-docs/reference/typescript/functions/htmlToMarkdown)

**Comment by AB1908** [comment]
What's the use case?

**Comment by Smitty010** [comment]
If you get the highlighted text from a book on an iPad Kindle app, it's an html file.  I wanted to parse that.  I eventually did that using regex.  The HTML that the ipad kindle app returns is pretty simple and predictable, so I think regex is adequate for now.  This should be closed.  I will also look at sohanglal's answer.

**Reply by AB1908** [reply]
Funnily enough, I actually have a Python script that does this. Feel free to adapt it: https://github.com/AB1908/kindleHTML-to-markdown

**Comment by Smitty010** [comment]
I will take a look.

Cheers

Scott
